### PR TITLE
Walkers: added method to set the walking function 

### PR DESCRIPTION
**Pull request by mikand**
This addresses #102

### CODE REVIEW

**Comment by marcogario** [comment]
What shall we do with calls to ```self.functions.clear()```? This refactoring shows that sub-classes do not need to access ```functions``` directly. Thus, the call to ```clear()``` needs to be wrapped.

The goal of using clear() and _is_complete, was to check that a new method had been defined for each node_type. However, this might be the wrong way of looking at this. We should rely on **testing** to guarantee that not only a method for each node has been defined, but also that each method makes sense.

Shall we remove the calls to ```clear()``` and deprecate ```is_complete()```? Can we do it in this branch?

File: pysmt/walkers/generic.py [hunk 1]
```diff
@@ -76,6 +77,20 @@ def __init__(self, env=None):
 
         return
 
+
+    def set_function(self, function, node_types):
```
Comment (marcogario) [inline pysmt/walkers/generic.py]: What about using ```*node_types```?

Comment (mikand) [inline pysmt/walkers/generic.py]: OK

**Comment by mikand** [comment]
@marcogario I agree that testing should look into this, but the is_complete method allows to catch subtle bugs early: it has the value of an assertion. Not sure if we really want to deprecate it...

**Pull request merged**